tests/rptest/tests/archival_test.py

```diff
@@ -205,7 +205,8 @@ def __init__(self, test_context):
 
         extra_rp_conf = dict(
             log_compaction_interval_ms=self.log_compaction_interval_ms,
-            log_segment_size=self.log_segment_size)
+            log_segment_size=self.log_segment_size,
+            retention_local_is_nonstrict=False)
```

Comment:
This commit makes non-strict mode enabled by default for new clusters. This is causing problems for tests that have checks like "n segments deleted" or that truncation occurs up to some point. I started disabling nonstrict mode in some tests, but found that this category of check is all over. So it seems like it might be better to update the tests to deal with the new default, but it's not yet clear what those updates will look like.

Reply:
A lot of cloud storage ducktape tests use the truncation of segments as an indicator that those segments have been uploaded to the cloud. The correct thing do to here is to infer this from the manifest uploaded in the bucket/container. We have a utility for this `quiesce_uploads` (in `si_utils.py`) and some newer tests are already using it.

Reply:
DIscussed offline.

Currently approach is to disable the feature for the cloud tests which utilize this older way of inferring cloud usage by examining that truncation is occurring. We'll need to go back and update these tests individually in order to get expanded test coverage with the feature enabled. For now we'll then be relying on existing tests designed for the feature to exercise it.